server/src/main/java/org/elasticsearch/action/search/SearchProgressListener.java

```diff
@@ -32,6 +32,15 @@ public abstract class SearchProgressListener {
 
     private List<SearchShard> shards;
 
+    /**
+     * Executed one when minimize round trips is enabled to indicate the number of remote clusters
+     * that will not be monitored during the search.
+     *
+     * @param hasLocalShards whether the search will include searching local shards
+     * @param numRemoteClusters how many remote clusters are part of the CCS search
+     */
+    protected void onMinimizeRoundtrips(boolean hasLocalShards, int numRemoteClusters) {}
```

Comment:
one thing I was wondering is if we need an additional callback, or we could rely on the existing onListShards. One reason why we may need it is that there is a case where you only search to a single remote cluster and no local indices, in which case I am not even sure onListShards is called. Maybe you also found other reasons.

Reply:
Yeah, that's why I marked it as WIP. I wanted to do a PR now to see if it passes the build (I forgot that I could do a draft PR until after I created it). I still have more research, testing and experimentation to do before going for a full PR review. Thanks for the early feedback!

Reply:
sounds good, this looks already pretty close, another aspect I would look into is whether we need additional testing or the duel test is enough.

Reply:
> One reason why we may need it is that there is a case where you only search to a single remote cluster and no local indices
I found a bug in this code for that exact case - where only querying a remote cluster. In that case, a call to notifyListShards was still necessary to initialize the searchResponse (MutableSearchResponse) in [AsyncSearchTask](https://github.com/elastic/elasticsearch/pull/96012/files#diff-24921eca27bf57c7b3d8d3ba7f02acbf6449d91e43e49bbd9df6377843adb19aR461).

So I've fixed that and added tests to CCSDuelIT that also tests a remote-cluster only (no local indices), which uncovers this bug and it now passes.